tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/SPContentTypes.cs

```diff
@@ -164,46 +199,69 @@ private void SetDefaultContentType(ClientContext clientContext, List list, Clien
+                    fields.GetByInternalNameOrTitle(contentTypesConfig.ContentTypeColumnMatterId).Update();
+                    fields.GetByInternalNameOrTitle(contentTypesConfig.ContentTypeColumnMatterName).DefaultValue = matterMetadata.Matter.Name;
+                    fields.GetByInternalNameOrTitle(contentTypesConfig.ContentTypeColumnMatterName).ReadOnlyField = true;
+                    fields.GetByInternalNameOrTitle(contentTypesConfig.ContentTypeColumnMatterName).SetShowInDisplayForm(true);
+                    fields.GetByInternalNameOrTitle(contentTypesConfig.ContentTypeColumnMatterName).Update();
 
-                int levels = taxonomySettings.Levels;
-                //For the number of levels that are configured, get the configured column name and 
-                //update the wssid, termname and id for the managed field
-                for (int i = 1; i <= levels; i++)
-                {
-                    string columnName = configuration.GetSection("ContentTypes").GetSection("ManagedColumns")["ColumnName" + i];
-                    fields.GetByInternalNameOrTitle(columnName).SetShowInDisplayForm(true);
-                    ManagedColumn managedColumn = matterMetadata.ManagedColumnTerms[columnName];
-                    fields.GetByInternalNameOrTitle(columnName).DefaultValue =
-                                string.Format(CultureInfo.InvariantCulture, ServiceConstants.MetadataDefaultValue,
-                                managedColumn.WssId,
-                                managedColumn.TermName,
-                                managedColumn.Id);
-                    fields.GetByInternalNameOrTitle(columnName).Update();
+                    int levels = taxonomySettings.Levels;
+                    //For the number of levels that are configured, get the configured column name and 
+                    //update the wssid, termname and id for the managed field
+                    for (int i = 1; i <= levels; i++)
+                    {
+                        string columnName = configuration.GetSection("ContentTypes").GetSection("ManagedColumns")["ColumnName" + i];
+                        fields.GetByInternalNameOrTitle(columnName).SetShowInDisplayForm(true);
+                        ManagedColumn managedColumn = matterMetadata.ManagedColumnTerms[columnName];
+                        fields.GetByInternalNameOrTitle(columnName).DefaultValue =
+                                    string.Format(CultureInfo.InvariantCulture, ServiceConstants.MetadataDefaultValue,
+                                    managedColumn.WssId,
+                                    managedColumn.TermName,
+                                    managedColumn.Id);
+                        fields.GetByInternalNameOrTitle(columnName).Update();
+                    }
+
+                    MatterExtraProperties matterExtraProperties = matterMetadata.MatterExtraProperties; ;

```

Comment:
This logic is to set the values of new content type site columns  and update it in sharepoint .
Update the values of site columns of Additional content type at Matter level